Hometask1

AlexanderKiselyov opened the pull request.

AlexanderKiselyov committed:
Create ITree.cs

AlexanderKiselyov committed:
Add files via upload

AlexanderKiselyov committed:
Create AssemblyInfo.cs

AlexanderKiselyov committed:
Add files via upload

AlexanderKiselyov committed:
Create TreeTests.cs

AlexanderKiselyov committed:
Add files via upload

AlexanderKiselyov committed:
Create AssemblyInfo.cs

yurii-litvinov commented on Homework 1/ITree.cs:
```diff
@@ -0,0 +1,27 @@
+		/// <summary>
+		/// Adds a value to the binary tree.
+		/// </summary>
+		/// <param name="value"></param>
```
Пустые описания параметров --- это плохо.

yurii-litvinov commented on Homework 1/Tree.cs:
```diff
@@ -0,0 +1,269 @@
+
+namespace Homework_1
+{
+	public partial class BinaryTree<T> : IBinaryTree<T> where T : IComparable<T>
```
А зачем он partial?

yurii-litvinov replied:
К каждому классу и каждому паблик-методу нужен комментарий.

yurii-litvinov commented on Homework 1/Tree.cs:
```diff
@@ -0,0 +1,269 @@
+				head = new TreeElement(value);
+				head.Parent = head;
+			}
+			else
```
Можно было в if-е сделать return, тогда else был бы не нужен, что сэкономило бы один уровень отступа всему коду ниже.

yurii-litvinov commented on Homework 1/Tree.cs:
```diff
@@ -0,0 +1,269 @@
+							return;
+						}
+						temp = temp.LeftChild;
+					}
```
Хм, а если элемент в дереве уже есть?

yurii-litvinov commented on Homework 1/Tree.cs:
```diff
@@ -0,0 +1,269 @@
+			return GetEnumerator();
+		}
+
+		public class Enumerator : IEnumerator<T>
```
А зачем он public?

yurii-litvinov commented on Homework 1/Tree.cs:
```diff
@@ -0,0 +1,269 @@
+
+		public class Enumerator : IEnumerator<T>
+		{
+			public BinaryTree<T> tree;
```
public-полей не бывает :)

yurii-litvinov commented on TreeTest/TreeTests.cs:
```diff
@@ -0,0 +1,37 @@
+				number++;
+			}
+			Assert.AreEqual(number, 13);
+		}
```
Тесты должны быть независимыми, чтобы если один не прошёл, стало бы понятно, в чём ошибка. И их должно быть много.

yurii-litvinov reviewed (changes requested):
Поправьте, пожалуйста.

AlexanderKiselyov committed:
Update ITree.cs

Добавлены комментарии к параметрам функций.

AlexanderKiselyov committed:
Update Tree.cs

Добавлены комментарии к public-методам и классам, исправлен баг с добавлением уже присутствующего в дереве элемента.

AlexanderKiselyov committed:
Update TreeTests.cs

Разделен один тест на несколько тестов, проверяющих функционал двоичного дерева.

yurii-litvinov commented on TreeTest/TreeTests.cs:
```diff
@@ -0,0 +1,92 @@
+			tree.AddElement(17);
+			tree.AddElement(17);
+			Assert.AreEqual(tree.SearchElement(3), false);
+			Assert.AreEqual(tree.SearchElement(11), true);
```
Это же Assert.IsTrue и Assert.IsFalse

yurii-litvinov commented on Homework 1/Tree.cs:
```diff
@@ -0,0 +1,307 @@
+						parent = temp;
+						temp = temp.LeftChild;
+					}
+				}
```
А зачем считать parent в цикле, если у узла есть свойство Parent?
